com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingRecursiveRenderer.cs

```diff
@@ -153,6 +167,54 @@ void RaytracingRecursiveRender(HDCamera hdCamera, CommandBuffer cmd, ScriptableR
             }
         }
 
+        class RecursiveRenderingPassData
+        {
+            public RecursiveRendererParameters parameters;
+            public TextureHandle depthStencilBuffer;
+            public TextureHandle flagMask;
+            public TextureHandle debugBuffer;
+            public TextureHandle rayCountTexture;
+            public TextureHandle outputBuffer;
+        }
+
+        TextureHandle RaytracingRecursiveRender(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle colorBuffer, TextureHandle depthPyramid, TextureHandle flagMask, TextureHandle rayCountTexture)
+        {
+            // If ray tracing is disabled in the frame settings or the effect is not enabled
+            RecursiveRendering recursiveSettings = hdCamera.volumeStack.GetComponent<RecursiveRendering>();
+            if (!hdCamera.frameSettings.IsEnabled(FrameSettingsField.RayTracing) || !recursiveSettings.enable.value)
+                return colorBuffer;
+
+            // Build the parameter structure
+            RecursiveRendererParameters rrParams = PrepareRecursiveRendererParameters(hdCamera, recursiveSettings);
+
+            using (var builder = renderGraph.AddRenderPass<RecursiveRenderingPassData>("Recursive Rendering Evaluation", out var passData, ProfilingSampler.Get(HDProfileId.RayTracingRecursiveRendering)))
+            {
+                builder.EnableAsyncCompute(false);
+
+                passData.parameters = rrParams;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.flagMask = builder.ReadTexture(flagMask);
+                passData.rayCountTexture = builder.ReadTexture(builder.WriteTexture(rayCountTexture));
+                passData.outputBuffer = builder.ReadTexture(builder.WriteTexture(colorBuffer));
+                passData.debugBuffer = builder.WriteTexture(renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
```

Comment:
Do we always write to the debugBuffer?
I suppose there should be a test here to not create the texture if it's not used.
As it stands, since the pass has other products (outputBuffer), if the output is used somewhere, the pass won't be culled so the debugBuffer will always be allocated as well.

Reply:
write a comment